### PR TITLE
Modify tests for testing objects for properties challenge

### LINKED ISSUE #38361: Testing Objects for Properties - Tests too weak
**Describe your problem and how to reproduce it:**
In https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/basic-javascript/testing-objects-for-properties, my wife made the test pass with if else condition for all keys in the javascript object.
I expect the tests should have a stronger test enforcement by testing for other objects in codebase. Would love to contribute to this one by myself

```function checkObj(checkProp) {
  // Only change code below this line
  if(checkProp == "gift") {
    return "pony";
  } else if(checkProp == "pet") {
    return "kitten";
  } else if(checkProp == "bed") {
    return "sleigh";
  } else {
    return "Not Found";
  }
  // Only change code above this line
}```

I'm teaching my wife to code with Freecodecamp, thanks a lot for all the effort in here :) 

**Add a Link to the page with the problem:**
https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/basic-javascript/testing-objects-for-properties

**Tell us about your browser and operating system:**
* Browser Name: Firefox
* Browser Version: 
* Operating System: Mac


**If possible, add a screenshot here (you can drag and drop, png, jpg, gif, etc. in this box):**

### PR DESCRIPTION
**Pull request by nkskalyan**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #38361

### CODE REVIEW

**Review by RandellDawson — CHANGES_REQUESTED** [review]
@nkskalyan You have at least one test failing that you need to fix.  Also, I suggest adding a space after the property name and colon for the objects in each `text`.  It makes it more readabel. For example:
```
  - text: '<code>checkObj({"city":" Seattle"}, "city")</code> should return  <code>"Seattle"</code>.'
```
Make sure to run `npm run test:curriculum` before pushing any commits, so you know for sure the tests pass with a valid solution.

**Comment by nkskalyan** [comment]
Done! Added space between colon. Had some issues with pupeteer, now have it fixed and tests are passing!

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/testing-objects-for-properties.english.md [hunk 1]
```diff
@@ -32,13 +32,16 @@ Modify the function <code>checkObj</code> to test <code>myObj</code> for <code>c
...
-  - text: <code>checkObj("house")</code> should return  <code>"Not Found"</code>.
-    testString: assert(checkObj("house") === "Not Found");
-
+  - text: '<code>checkObj({gift: "pony", pet: "kitten", bed: "sleigh"}, "gift")</code> should return  <code>"pony"</code>.'
+    testString: 'assert(checkObj({gift: "pony", pet: "kitten", bed: "sleigh"}, "gift") === "pony");'
+  - text: '<code>checkObj({gift: "pony", pet: "kitten", bed: "sleigh"}, "pet")</code> should return  <code>"kitten"</code>.'
+    testString: 'assert(checkObj({gift: "pony", pet: "kitten", bed: "sleigh"}, "pet") === "kitten");'
+  - text: '<code>checkObj({gift: "pony", pet: "kitten", bed: "sleigh"}, "house")</code> should return  <code>"Not Found"</code>.'
+    testString: 'assert(checkObj({gift: "pony", pet: "kitten", bed: "sleigh"}, "house") === "Not Found");'
+  - text: '<code>checkObj({"city": "Seattle"}, "city")</code> should return  <code>"Seattle"</code>.'
```
Comment (RandellDawson) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/testing-objects-for-properties.english.md]:
```suggestion
  - text: '<code>checkObj({city: "Seattle"}, "city")</code> should return <code>"Seattle"</code>.'
```

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/testing-objects-for-properties.english.md [hunk 2]
```diff
@@ -32,13 +32,16 @@ Modify the function <code>checkObj</code> to test <code>myObj</code> for <code>c
...
 ```yml
 tests:
-  - text: <code>checkObj("gift")</code> should return  <code>"pony"</code>.
-    testString: assert(checkObj("gift") === "pony");
-  - text: <code>checkObj("pet")</code> should return  <code>"kitten"</code>.
-    testString: assert(checkObj("pet") === "kitten");
-  - text: <code>checkObj("house")</code> should return  <code>"Not Found"</code>.
-    testString: assert(checkObj("house") === "Not Found");
-
+  - text: '<code>checkObj({gift: "pony", pet: "kitten", bed: "sleigh"}, "gift")</code> should return  <code>"pony"</code>.'
```
Comment (RandellDawson) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/testing-objects-for-properties.english.md]:
```suggestion
  - text: '<code>checkObj({gift: "pony", pet: "kitten", bed: "sleigh"}, "gift")</code> should return <code>"pony"</code>.'
```

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/testing-objects-for-properties.english.md [hunk 3]
```diff
@@ -32,13 +32,16 @@ Modify the function <code>checkObj</code> to test <code>myObj</code> for <code>c
...
+    testString: 'assert(checkObj({gift: "pony", pet: "kitten", bed: "sleigh"}, "gift") === "pony");'
+  - text: '<code>checkObj({gift: "pony", pet: "kitten", bed: "sleigh"}, "pet")</code> should return  <code>"kitten"</code>.'
```
Comment (RandellDawson) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/testing-objects-for-properties.english.md]:
```suggestion
  - text: '<code>checkObj({gift: "pony", pet: "kitten", bed: "sleigh"}, "pet")</code> should return <code>"kitten"</code>.'
```

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/testing-objects-for-properties.english.md [hunk 4]
```diff
@@ -32,13 +32,16 @@ Modify the function <code>checkObj</code> to test <code>myObj</code> for <code>c
...
+    testString: 'assert(checkObj({gift: "pony", pet: "kitten", bed: "sleigh"}, "pet") === "kitten");'
+  - text: '<code>checkObj({gift: "pony", pet: "kitten", bed: "sleigh"}, "house")</code> should return  <code>"Not Found"</code>.'
```
Comment (RandellDawson) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/testing-objects-for-properties.english.md]:
```suggestion
  - text: '<code>checkObj({gift: "pony", pet: "kitten", bed: "sleigh"}, "house")</code> should return <code>"Not Found"</code>.'
```

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/testing-objects-for-properties.english.md [hunk 5]
```diff
@@ -32,13 +32,16 @@ Modify the function <code>checkObj</code> to test <code>myObj</code> for <code>c
...
+    testString: 'assert(checkObj({gift: "pony", pet: "kitten", bed: "sleigh"}, "house") === "Not Found");'
+  - text: '<code>checkObj({"city": "Seattle"}, "city")</code> should return  <code>"Seattle"</code>.'
+    testString: 'assert(checkObj({"city": "Seattle"}, "city") === "Seattle");'
```
Comment (RandellDawson) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/testing-objects-for-properties.english.md]:
```suggestion
    testString: 'assert(checkObj({city: "Seattle"}, "city") === "Seattle");'
```

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/testing-objects-for-properties.english.md [hunk 6]
```diff
@@ -32,13 +32,16 @@ Modify the function <code>checkObj</code> to test <code>myObj</code> for <code>c
...
+  - text: '<code>checkObj({"city": "Seattle"}, "district")</code> should return  <code>"Not Found"</code>.'
```
Comment (RandellDawson) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/testing-objects-for-properties.english.md]:
```suggestion
  - text: '<code>checkObj({city: "Seattle"}, "district")</code> should return <code>"Not Found"</code>.'
```

**Review by RandellDawson — CHANGES_REQUESTED** [review]
Please see the following suggested changes.  You had an extra space before the word `return` in the `text`.  Also, there is no reason to put double quotes around the `city` property name in the last two tests.

FYI -  You can simply click `Commit suggestion` for each suggestion and add an applicable commit message instead of creating a new commit locally with all the changes.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/testing-objects-for-properties.english.md (hunk 6 above)
Comment (nkskalyan) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/testing-objects-for-properties.english.md]: Removed all the double spaces from return, removed double quotes in in object key

**Review by RandellDawson — APPROVED** [review]

**Review by Manish-Giri — CHANGES_REQUESTED** [review]
Since the function `checkProp` has been changed to include an object as the first parameter, `obj`, which must now be tested, instead of the original object `myObj`, the challenge instruction should also be changed accordingly -

> Modify the function `checkObj` to test `myObj` for `checkProp`

should be

> Modify the function `checkObj` to test `obj` for `checkProp`

Otherwise, it looks like we want campers to check for the `myObj` object included in the seed code -
```
var myObj = {
  gift: "pony",
  pet: "kitten",
  bed: "sleigh"
};
```
Since `myObj` doesn't have a `city` property, the new test wouldn't make much sense -
> `checkObj({city: "Seattle"}, "city")` should return "Seattle"

**Comment by RandellDawson** [comment]
@Manish-Giri Good catch!  What about the following for the instructions?
> Modify the function <code>checkObj</code> to test if an object passed to the function (<code>obj</code>) contains a specific property (<code>checkProp</code>). If the property is found, return that property's value. If not, return <code>"Not Found"</code>.

**Comment by Manish-Giri** [comment]
@RandellDawson Yes, that works 💯

**Comment by RandellDawson** [comment]
@nkskalyan Can you please add another commit to incorporate the following for the instructions section agreed upon by Manish and me?  Thanks.
```
<section id='instructions'>
Modify the function <code>checkObj</code> to test if an object passed to the function (<code>obj</code>) contains a specific property (<code>checkProp</code>). If the property is found, return that property's value. If not, return <code>"Not Found"</code>.
</section>
```

**Review by RandellDawson — APPROVED** [review]

**Comment by RandellDawson** [comment]
@Manish-Giri I think this looks good now.

**Review by Manish-Giri — APPROVED** [review]

**Comment by RandellDawson** [comment]
Congrats @nkskalyan on your first pull request (PR)! 🎉

Thank you for your contribution and patience through the process! 👍
We're happy to accept these changes, and look forward to future contributions. 📝

**Comment by nkskalyan** [comment]
Thanks Randell for timely and useful feedback! Excited to contribute more in future

**Pull request merged**